src/subscribers/websocket.c

```diff
@@ -916,33 +927,36 @@ ngx_int_t websocket_subscriber_destroy(subscriber_t *sub) {
   return NGX_OK;
 }
 
-static ngx_int_t extract_deflate_window_bits(full_subscriber_t *fsub, u_char *lcur, u_char *lend, const char* setting_name, int8_t *dst, int min, int max) {
-  int        bits= -1;
+static ngx_int_t extract_deflate_window_bits(full_subscriber_t *fsub, u_char *lcur, u_char *lend, const char* setting_name) {
+  ngx_int_t bits= -1;
   u_char    *ltmp;
   if((ltmp = ngx_strnstr(lcur, (char *)setting_name, lend - lcur)) != NULL) {
     ltmp += strlen(setting_name); //strlen
     if(*ltmp == '=') ltmp++;
     if(*ltmp == '"') ltmp++;
     u_char    *nend = ltmp;
     while(nend <= lend && *nend >='0' && *nend <='9') nend++;
-    bits = ngx_atoi(ltmp, nend - ltmp);
-    if(bits == NGX_ERROR) {
-      // the client is not limiting client_max_window_bits, so we must use the max
-      // this is in accordance with RFC 7692: the client may announce the extension
-      // without specifying a value
-      // for server_max_window_bits, this shouldn't happen
-      bits = 15;
+    if(nend == lend) {
```

Comment:
This incorrectly ignores numbers-of-bits at the end of the line. The correct code is
```
    if(nend - ltmp == 0) {
      //no value, don't set
      return NGX_OK;
    }
```

(no need to update the PR, i'm fixing t in the merge)